cwl/src/test/scala/wdl4s/cwl/ExportCwlSamplesSpec.scala

```diff
@@ -28,27 +39,166 @@ class ExportCwlSamplesSpec extends FlatSpec with Matchers {
           `type` = None
         ))),
         outputs = Coproduct[CommandLineTool.Outputs](Array.empty[CommandOutputParameter]),
-        `class` = "CommandLineTool",
         id = None,
         requirements = None,
         hints = None,
         label = None,
         doc = None,
         cwlVersion = Option(CwlVersion.Version1),
-        baseCommand = None,
+        baseCommand = Option(Coproduct[BaseCommand]("echo")),
         arguments = None,
         stdin = None,
         stderr = None,
         stdout = None,
         successCodes = None,
         temporaryFailCodes = None,
         permanentFailCodes = None)
-    val toolJson = encodeCwlCommandLineTool(tool)
-    val printer = new Printer(preserveOrder = true, dropNullKeys = true, indent = "  ")
-    val toolJsonString = printer.pretty(toolJson)
     val expectedToolJsonString =
-      """{"inputs":{"message":{"inputBinding":{"position":1}}},"outputs":[],"class":"CommandLineTool","cwlVersion":"v1.0"}"""
-    toolJsonString shouldBe expectedToolJsonString
+      """{
+        |  "inputs" : {
+        |    "message" : {
+        |      "inputBinding" : {
+        |        "position" : 1
+        |      }
+        |    }
+        |  },
+        |  "outputs" : [
+        |  ],
+        |  "class" : "CommandLineTool",
+        |  "cwlVersion" : "v1.0",
+        |  "baseCommand" : "echo"
+        |}""".stripMargin
+    assertCorrectJson(tool, expectedToolJsonString)
+  }
+
+  it should "encode sample CWL workflow" in {
+    val workflow = Workflow(
+      cwlVersion = Option(CwlVersion.Version1),
+//      `class` = "Workflow",
+      inputs = Coproduct[WorkflowInput](
```

Comment:
Agree the `Coproduct` is noisy.  In some cases the type argument isn't necessary but I can't figure out when the compiler needs it and when it doesn't.

In any case I haven't figured out how to implicitly create all these `Coproduct`s.